crypto/bio/bio_local.h

```diff
@@ -134,6 +134,12 @@ struct bio_st {
     uint64_t num_write;
     CRYPTO_EX_DATA ex_data;
     CRYPTO_RWLOCK *lock;
+#ifndef OPENSSL_NO_SOCK
+    struct {
+        BIO_ADDR peer;
+        int first;
+    } tfo;
+#endif
```

Comment:
Should we really be putting this in the top level bio_st? Wouldn't this be better placed in a BIO specific structure?

Reply:
Can't use `ptr` and the other option is `ex_data`, both of which are user visible, and `ex_data` use is challenging enough that it's easy for a user to break things. I originally had this data stored in `ptr`, but because it's accessible via `BIO_get_data()` it broke some user apps... they should've used `ex_data`, but that's hard to use correctly. I'm actually a little concerned that the `BIO_F_BUFFER_CTX` structure is stored in `ptr`...

This is a fairly small amount of data, and I don't think it's that burdensome. I would want to add a new pointer (since I can't use the user-visible `ptr`) and it subsequently becomes the difference between `sizeof(void*)` vs `sizeof(struct sockaddr) + sizeof(int)`

Reply:
I realize I'm harping on `ptr`, but it bit us pretty bad!

Reply:
Fixed